### PR TITLE
update by symonhay

### PR DESCRIPTION
**Pull request by miraculixx**

### CODE REVIEW

File: src/android/GeofenceTransitionsService.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    }
+
+
+    @Override
+    public void onCreate(){ super.onCreate(); intent = new Intent(TAG); }
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        initializeLocationManager();
+
+        GeofencingEvent geofencingEvent = GeofencingEvent.fromIntent(intent);
```
Comment (miraculixx) [inline src/android/GeofenceTransitionsService.java]: this doesn't look like it's tested at least not thoroughly :-1: . For some reason you keep mixing up geofencing and location updates. Again, these are not the same thing:
- location updates are sent _every time_ the phone registers a new location. for this you need the LocationManager, which needs to be called from a Service that keeps running in the background because the LocationManager needs a live LocationListener to receive updates. this is _independent of geofencing_
- geofences are _only_ sent when a defined region boundary is crossed. For this you need a IntentService because updates are delivered as intents. 

You have implemented it somewhat the other way around, why? 
- MultiRegionService is now an IntentService, but it is started as a service. If this works at all, it means the MultiRegionService dies immediately after it has finished its `onHandleIntent` method, while it should keep running to receive LocationListener callbacks.
- GeofenceTransitionService, which should be an IntentService, is now a Service. This means GeofenceTransitionService will keep running while  it should really just handle geofencing updates and then die.

While I doubt this will work (at least reliably) in the first place, what's worse the **original problem is still not solved**: location updates are only started when the first geofence transition has occured, which may be never. That's wrong. Location updates shall be received independent of whether the user has ever crossed a geofence boundary.

In summary, all code is there but it's in the wrong place and it has the semantics of geofencing and location updates mixed up. Why?

The correct order IMO is:
1. start a service to request location updates via LocationManager. This needs to happen in the context of a service so that the LocationListener is alive when the callback happens. Why not use MultiRegionService for that?
2. add geofencing and register the IntentService to handle geofencing events. You had it implemented this way in a previous submission, why change it? The only thing that was wrong is that the location manager was initialized in the wrong place, and the LocationListener was incorrectly attached to the geofencing IntentService.

**Comment by miraculixx** [comment]
please also see this https://github.com/miraculixx/cordova-rfscanner-plugin/commit/9cc86206d4c6369df97be0d8bb101cbd30951e63#commitcomment-16468257
